src/pudl/transform/eia861.py

```diff
@@ -1020,6 +1014,86 @@ def _harvest_associations(dfs: list[pd.DataFrame], cols: list[str]) -> pd.DataFr
+    values are known withholdings whereas the duplicate primary key rows are unknown data
+    quality issues.
+
+    Args:
+        df: The dataframe to process.
+        idx_cols: The index (primary key) columns of the input dataframe. They must
+            identify unique records in the input data.
+
+    Returns:
+        A dataframe with the combined or dropped 88888 utiltiy_id_eia values and unique pks.
+    """
+    if 88888 not in df.utility_id_eia.unique():
+        return df
+    non_num_cols = (
+        df.set_index(idx_cols).select_dtypes(exclude="number").columns.tolist()
+    )
+
+    def custom_group_agg(group):
+        if len(group) <= 1:
+            return group
+        # If there are no non-numeric columns, or all non-numeric columns have the same value,
+        # Sum the numeric columns and return the result.
+        if (
+            len(non_num_cols) == 0
+            or (group[non_num_cols] == group[non_num_cols].iloc[0]).all().all()
+        ):
+            # Special case: if there are NA values in the index columns we can't
+            # combine two dfs with the same index without creating two rows.
+            # This messes up the results, so we drop the NA values from the index
+            # when combining the newly aggregated groups. This does not affect the
+            # actual primary key, it's just a slightly hacky way to combine the rows correctly.
+            idx_has_na = group[idx_cols].isna().any()
+            na_cols = idx_has_na[idx_has_na].index.tolist()
+            mod_idx_cols = [x for x in idx_cols if x not in na_cols]
+            # Now we can safely sum the numeric columns, recombine with the non-numeric
+            # columns, and return the result.
+            group = group.set_index(mod_idx_cols)
+            non_num_group = group[non_num_cols].iloc[[0]]
+            num_group = (
+                group.drop(columns=non_num_cols)
+                .groupby(mod_idx_cols, dropna=False)
+                .sum()
+            )
+            return pd.concat([non_num_group, num_group], axis="columns").reset_index()
+        # When there are rows that can't be combined due to discrepencies in the non-numeric
+        # columns, we drop the duplicates and return the first row.
+        # This is a bit of a hack, but it ensures that we don't end up with multiple rows
+        # with the same primary key, which would cause problems later on.
+        return group.drop_duplicates(subset=idx_cols)
+
+    utils_88888 = df[df["utility_id_eia"] == 88888]
+    agg_utils_88888 = utils_88888.groupby(
+        idx_cols, group_keys=False, dropna=False
+    ).apply(custom_group_agg)
+    recombined_df = pd.concat(
+        [df[df["utility_id_eia"] != 88888], agg_utils_88888], ignore_index=True
+    )
+    # Make sure that the number of rows altered stays somewhat small. We don't expect
+    # there to be a lot of dropped or combined 88888 rows.
+    if (len(df) - len(recombined_df)) > 35:
```

Comment:
blocking: Might be good for this to be a ratio instead of a flat number - that'll be a much more meaningful definition of 'somewhat small.'